app/connectors/PdsAuthCheckerConnector.scala

```diff
@@ -17,46 +17,64 @@
-import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, StringContextOps, UpstreamErrorResponse}
+import uk.gov.hmrc.http.HttpReads.Implicits._
 import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{
+  HeaderCarrier,
+  HttpResponse,
+  StringContextOps,
+  UpstreamErrorResponse
+}
 import uk.gov.hmrc.ukimauthcheckerapi.config.AppConfig
-import play.api.libs.json.OFormat.oFormatFromReadsAndOWrites
+
 import scala.concurrent.{ExecutionContext, Future}
 
 @ImplementedBy(classOf[PdsAuthCheckerConnectorImpl])
 trait PdsAuthCheckerConnector {
-  def check(request: AuthorisationRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[PdsAuthCheckerResponse]
+  def check(request: AuthorisationRequest)(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext
+  ): Future[Either[ValidationErrorResponse, PdsAuthCheckerResponse]]
 }
 
 @Singleton
-class PdsAuthCheckerConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig: AppConfig) extends PdsAuthCheckerConnector {
-  def check(request: AuthorisationRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[PdsAuthCheckerResponse] = {
+class PdsAuthCheckerConnectorImpl @Inject() (
+    httpClientV2: HttpClientV2,
+    appConfig: AppConfig
+) extends PdsAuthCheckerConnector {
+  def check(request: AuthorisationRequest)(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext
+  ): Future[Either[ValidationErrorResponse, PdsAuthCheckerResponse]] = {
 
     val authType = "UKIM"
     val url = appConfig.pdsAuthCheckerUrl.addPathParts("authorisations")
-    val pdsRequest = PdsAuthCheckerRequest(request.date, authType, request.eoris)
+    val pdsRequest =
+      PdsAuthCheckerRequest(request.date, authType, request.eoris)
 
-    httpClientV2.post(url"$url")
+    httpClientV2
+      .post(url"$url")
       .withBody(Json.toJson(pdsRequest))
       .execute[HttpResponse]
-      .flatMap {
-        response =>
-          response.status match {
-            case OK => {
-              response.json
-                .validate[PdsAuthCheckerResponse]
-                .map(
-                  result => Future.successful(result)
-                )
-                .recoverTotal(
-                  error => Future.failed(JsResult.Exception(error))
-                )
-            }
-            case _ => Future.failed(UpstreamErrorResponse(response.body, response.status))
-          }
+      .flatMap { response =>
+        response.status match {
+          case OK =>
+            response.json
+              .validate[PdsAuthCheckerResponse]
+              .map(result => Future.successful(Right(result)))
+              .recoverTotal(error => Future.failed(JsResult.Exception(error)))
+          case BAD_REQUEST =>
+            response.json
+              .validate[ValidationErrorResponse]
+              .map(result => Future.successful(Left(result)))
+              .recoverTotal(error => Future.failed(JsResult.Exception(error)))
```

Comment:
Not vital, but there is an option here to refactor this to avoid duplication (validate on [A], map function as a parameter, always has the recoverTotal)

Reply:
Had another look at this, couldn't figure out a simple way to write this function as we'd need need to know both sides of the either, and which one we are mapping (for the left or the right), which I cannot think of a way of doing, happy to hear suggestions if i've gone about this the wrong way though (Maybe add in a new ticket to look at 😄)